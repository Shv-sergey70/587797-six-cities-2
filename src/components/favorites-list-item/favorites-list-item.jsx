import React from 'react';
import PropTypes from 'prop-types';
import FavoritesItemOffer from "../favorites-item-offer/favorites-item-offer";
import offersPropTypes from '../../prop-types/offer';

const FavoritesListItem = (props) => {
  const {
    cityName,
    offers
  } = props;

  return <li className="favorites__locations-items">
    <div className="favorites__locations locations locations--current">
      <div className="locations__item">
        <a className="locations__item-link" href="#">
          <span>{cityName}</span>
        </a>
      </div>
    </div>
    <div className="favorites__places">
      {offers.map((offer) => <FavoritesItemOffer
        key={`${cityName}-${offer.id}`}
        offer={offer}
      />
      )}
    </div>
  </li>;
};

FavoritesListItem.propTypes = {
  cityName: PropTypes.string.isRequired,
  offers: PropTypes.arrayOf(offersPropTypes)
};

export default FavoritesListItem;
