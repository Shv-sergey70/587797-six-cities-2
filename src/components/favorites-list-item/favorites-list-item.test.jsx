import React from 'react';
import renderer from 'react-test-renderer';
import {FavoritesListItem} from "./favorites-list-item";

jest.mock(`../favorites-item-offer/favorites-item-offer.jsx`, () => `FavoritesItemOffer`);

const offersData = [
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: true,
    type: `apartment`,
    price: 80,
    rating: 4.4,
    bedrooms: 2,
    maxAdults: 2,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/studio-01.jpg`],
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    description: `A quiet cozy and picturesque that hides behind a a river`
  },
  {
    id: 2,
    title: `Beautiful & luxurious apartment at great location2`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: false,
    isFavorite: true,
    type: `apartment`,
    price: 180,
    rating: 2.4,
    bedrooms: 5,
    maxAdults: 3,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/studio-01.jpg`],
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`],
    description: `A quiet cozy and picturesque that hides behind a a river1`
  },
  {
    id: 3,
    title: `Beautiful & luxurious apartment at great location33`,
    previewImage: `/img/apartment-01.jpg`,
    isPremium: true,
    isFavorite: false,
    type: `room`,
    price: 820,
    rating: 4.9,
    bedrooms: 5,
    maxAdults: 4,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`],
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`],
    description: `A quiet cozy and picturesque that hides behind a a river22`
  }
];

it(`FavoritesListItem correctly renders`, () => {
  const tree = renderer.create(
      <FavoritesListItem
        cityName={`Amsterdam`}
        offers={offersData}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
