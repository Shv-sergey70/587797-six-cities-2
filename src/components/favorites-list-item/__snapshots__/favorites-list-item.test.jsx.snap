// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesListItem correctly renders 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <FavoritesItemOffer
      offer={
        Object {
          "bedrooms": 2,
          "description": "A quiet cozy and picturesque that hides behind a a river",
          "goods": Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
            "Dishwasher",
          ],
          "id": 1,
          "images": Array [
            "/img/room.jpg",
            "/img/apartment-01.jpg",
            "/img/studio-01.jpg",
          ],
          "isFavorite": true,
          "isPremium": true,
          "maxAdults": 2,
          "previewImage": "/img/apartment-01.jpg",
          "price": 80,
          "rating": 4.4,
          "title": "Beautiful & luxurious apartment at great location",
          "type": "apartment",
        }
      }
    />
    <FavoritesItemOffer
      offer={
        Object {
          "bedrooms": 5,
          "description": "A quiet cozy and picturesque that hides behind a a river1",
          "goods": Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
            "Coffee machine",
          ],
          "id": 2,
          "images": Array [
            "/img/room.jpg",
            "/img/apartment-01.jpg",
            "/img/studio-01.jpg",
          ],
          "isFavorite": true,
          "isPremium": false,
          "maxAdults": 3,
          "previewImage": "/img/apartment-01.jpg",
          "price": 180,
          "rating": 2.4,
          "title": "Beautiful & luxurious apartment at great location2",
          "type": "apartment",
        }
      }
    />
    <FavoritesItemOffer
      offer={
        Object {
          "bedrooms": 5,
          "description": "A quiet cozy and picturesque that hides behind a a river22",
          "goods": Array [
            "Heating",
            "Kitchen",
            "Cable TV",
            "Washing machine",
          ],
          "id": 3,
          "images": Array [
            "/img/room.jpg",
            "/img/apartment-01.jpg",
          ],
          "isFavorite": false,
          "isPremium": true,
          "maxAdults": 4,
          "previewImage": "/img/apartment-01.jpg",
          "price": 820,
          "rating": 4.9,
          "title": "Beautiful & luxurious apartment at great location33",
          "type": "room",
        }
      }
    />
  </div>
</li>
`;
